packages/appium/lib/utils.js

```diff
@@ -34,18 +33,42 @@ function inspectObject (args) {
   }
 }
 
-function parseExtensionArgs (extensionArgs, extensionName) {
-  if (!_.isString(extensionArgs)) {
+function getExtensionArgs (extensionArgs, extensionName) {
+  if (!_.has(extensionArgs, extensionName)) {
     return {};
   }
-  const parsedExtensionArgs = parseJsonStringOrFile(extensionArgs);
-  const extensionSpecificArgs = parsedExtensionArgs[extensionName];
-  if (!_.isPlainObject(extensionSpecificArgs)) {
+  if (!_.isPlainObject(extensionArgs[extensionName])) {
     throw new Error(`Driver or plugin arguments must be plain objects`);
   }
-  return extensionSpecificArgs;
+  return extensionArgs[extensionName];
 }
 
+
+function ensureNoUnknownArgs (extensionArgs, argsConstraints) {
+  const knownArgNames = Object.keys(argsConstraints);
+  for (const argName of Object.keys(extensionArgs)) {
+    if (!knownArgNames.includes(argName)) {
+      throw new Error(`"${argName}" is not a recognized arg. Are you sure it's in the list ` +
+                      `of supported args? ${JSON.stringify(knownArgNames)}`);
+    }
+  }
+}
+
+/**
+ * Takes in a set of driver/plugin args passed in by user, and arg constraints
+ * and throws an error if any arg is unknown or of the incorrect type
+ *
+ * @param {object} extensionArgs - Driver or Plugin specific args
+ * @param {object} argsConstraints - Constraints for arguments
+*/
+function validateExtensionArgs (extensionArgs, argsConstraints) {
+  if (!_.isEmpty(extensionArgs) && !_.isEmpty(argsConstraints)) {
+    ensureNoUnknownArgs(extensionArgs, argsConstraints);
+    validateCaps(extensionArgs, argsConstraints);
```

Comment:
we could probably rename this method on import to make it more readable, e.g. `import {validateCaps as validateArgs}`